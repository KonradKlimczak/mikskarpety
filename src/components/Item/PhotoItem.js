import React, { Component } from "react";

class PhotoItem extends Component {
    render() {
        return (
            <figure>
                <img className="big-grid" src={this.props.src} />
                {this.props.caption && <figcaption>{this.props.caption}</figcaption>}
            </figure>
        );
    }
}

export default PhotoItem;
