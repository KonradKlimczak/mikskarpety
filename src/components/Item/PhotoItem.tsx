import classnames from 'classnames';
import * as React from 'react';

import { getSizeClassName, Size } from '../../utils/Size';

interface IPhotoItemProps {
  src: string;
  caption?: string;
  size?: Size;
}

const PhotoItem: React.SFC<IPhotoItemProps> = (props) => {
  return (
    <figure className={classnames('photo-item', getSizeClassName(props.size))} tabIndex={0}>
      <img src={props.src} />
      {props.caption && <figcaption>{props.caption}</figcaption>}
    </figure>
  );
};

export default PhotoItem;
