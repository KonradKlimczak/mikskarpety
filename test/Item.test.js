import React from "react";
import renderer from "react-test-renderer";

import PhotoItem from "../src/components/Item/PhotoItem";

it("PhotoItem renders correctly", () => {
  const tree = renderer
    .create(<PhotoItem src="images/pictures/w01.jpg" caption="TEST" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});