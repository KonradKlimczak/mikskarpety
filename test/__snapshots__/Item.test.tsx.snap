// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoItem renders correctly 1`] = `
<figure
  className="photo-item size--medium"
  tabIndex={0}
>
  <img
    src="images/pictures/w01.jpg"
  />
  <figcaption>
    TEST
  </figcaption>
</figure>
`;
